tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationSettings.cs

```diff
@@ -7,36 +7,29 @@
 
 using System;
 using Datadog.Trace.Configuration;
+using Datadog.Trace.Configuration.Telemetry;
+using Datadog.Trace.Telemetry;
 
 namespace Datadog.Trace.RemoteConfigurationManagement
 {
     internal class RemoteConfigurationSettings
     {
         private const int DefaultPollIntervalMilliseconds = 5000;
 
-        public RemoteConfigurationSettings()
-            : this(configurationSource: null)
-        {
-        }
-
-        public RemoteConfigurationSettings(IConfigurationSource? configurationSource)
+        public RemoteConfigurationSettings(IConfigurationSource? configurationSource, IConfigurationTelemetry telemetry)
         {
             configurationSource ??= NullConfigurationSource.Instance;
 
             Id = Guid.NewGuid().ToString();
             RuntimeId = Util.RuntimeId.Get();
             TracerVersion = TracerConstants.ThreePartVersion;
 
-            var pollInterval =
-                configurationSource.GetInt32(ConfigurationKeys.Rcm.PollInterval)
+            var pollInterval = new ConfigurationBuilder(configurationSource, telemetry)
 #pragma warning disable CS0618
-                    ?? configurationSource.GetInt32(ConfigurationKeys.Rcm.PollIntervalInternal);
+                              .WithKeys(ConfigurationKeys.Rcm.PollInterval, ConfigurationKeys.Rcm.PollIntervalInternal)
 #pragma warning restore CS0618
-
-            pollInterval =
-                pollInterval is null or <= 0 or > 5000
-                    ? DefaultPollIntervalMilliseconds
-                    : pollInterval.Value;
+                              .AsInt32()
+                              .Get(DefaultPollIntervalMilliseconds, pollInterval => pollInterval is <= 0 or > 5000);
```

Comment:
It should be the opposite (between 0 and 5000)

Reply:
:oops:

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/4033/commits/552b46c42dbc07f27f85641d571487978026695b

Reply:
I assume it means we're missing one UTest. if you agree, would you mind adding it Andrew ?

Reply:
We're missing a lot more than one UTest - a _lot_ of these are never explicitly tested 😄 Kevin has taken it upon himself to add a bunch of tests for config that we can add _before_ this PR, so I can rebase on top and confirm we've not broken anything else 🙂

Reply:
Obv we miss a lot of UTests. It's just that now I know about this one :). 
I'm ok with the best strategy you think relevant (maybe rebasing this on top of Kevin's PR isn't the best idea if we're not to merge Kevin's PR soon?). I'll let you too decide what's the best way forward.